app/src/main/java/com/duckduckgo/app/onboarding/ui/page/WelcomePage.kt

```diff
@@ -95,17 +94,23 @@ class WelcomePage : OnboardingPageFragment(R.layout.content_onboarding_welcome)
                 .flowOn(dispatcherProvider.io())
                 .collect(::render)
         }
+
+        requestNotificationsPermissions()
     }
 
-    @SuppressLint("InlinedApi")
     private fun requestNotificationsPermissions() {
         if (appBuildConfig.sdkInt >= android.os.Build.VERSION_CODES.TIRAMISU) {
-            requestPermission.launch(Manifest.permission.POST_NOTIFICATIONS)
+            event(WelcomePageView.Event.OnNotificationPermissionsRequested)
         } else {
             scheduleWelcomeAnimation()
         }
     }
 
+    @SuppressLint("InlinedApi")
+    private fun showNotificationsPermissionsPrompt() {
+        requestPermission.launch(Manifest.permission.POST_NOTIFICATIONS)
```

Comment:
This method only works in TIRAMISU, we should be checking that here. I know if I follow the chain up, it's checked there but it doesn't stop anyone to directly call this without knowing what it does internally.